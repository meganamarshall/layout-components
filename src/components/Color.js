import React from 'react';

const hexFromColor = color => {
  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
};

const hexFromRGB = (rgb) => {
  return `#${hexFromColor(rgb.red)}${hexFromColor(rgb.green)}${hexFromColor(rgb.blue)}`;
};

export default function Color() {
  const name = 'Red';
  const rgb = {
    red: 255,
    green: 0,
    blue: 0
  };
  const hex = hexFromRGB(rgb);
  
  const style = {
    backgroundColor: hex,
    display: 'inline-block',
    width: '5px',
    height: '5px'
  };

  return (
    <dl style={style}>
      <dt>Name:</dt>
      <dd>{name}</dd>

      <dt>Hex:</dt>
      <dd>{hex}</dd>

      <dt>RGB:</dt>
      <dd>
        <p>Red - {rgb.red}</p>
        <p>Green - {rgb.green}</p>
        <p>Blue - {rgb.blue}</p>
      </dd>
    </dl>
  );
}
