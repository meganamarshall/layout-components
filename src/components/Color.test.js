import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('matches color snapshot', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
