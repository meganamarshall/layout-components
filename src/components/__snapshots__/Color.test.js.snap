// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component matches color snapshot 1`] = `
<dl
  style={
    Object {
      "backgroundColor": "#FF0000",
      "display": "inline-block",
      "height": "5px",
      "width": "5px",
    }
  }
>
  <dt>
    Name:
  </dt>
  <dd>
    Red
  </dd>
  <dt>
    Hex:
  </dt>
  <dd>
    #FF0000
  </dd>
  <dt>
    RGB:
  </dt>
  <dd>
    <p>
      Red - 
      255
    </p>
    <p>
      Green - 
      0
    </p>
    <p>
      Blue - 
      0
    </p>
  </dd>
</dl>
`;
